WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -0,0 +1,67 @@
+import UIKit
+import Yosemite
+
+/// Controls navigation for the flow to add a product given a navigation controller.
+///
+final class AddProductCoordinator: Coordinator {
+    var navigationController: UINavigationController
+
+    private let siteID: Int64
+    private let sourceView: UIBarButtonItem
+
+    init(siteID: Int64, sourceView: UIBarButtonItem, sourceNavigationController: UINavigationController) {
+        self.siteID = siteID
+        self.sourceView = sourceView
+        self.navigationController = sourceNavigationController
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func start() {
+        presentProductTypeBottomSheet()
+    }
+}
+
+// MARK: Navigation
+private extension AddProductCoordinator {
+    func presentProductTypeBottomSheet() {
+        let title = NSLocalizedString("Select a product type",
+                                      comment: "Message title of bottom sheet for selecting a product type to create a product")
+        let viewProperties = BottomSheetListSelectorViewProperties(title: title)
+        let command = ProductTypeBottomSheetListSelectorCommand(selected: nil) { selectedProductType in
+            self.navigationController.dismiss(animated: true) {
+                // Strong reference to `self` is required since `AddProductCoordinator` is not strongly referenced by any class.
```

Comment:
Maybe we should add a comment on the top of this class to warn the consumer that this class should not be retained, WDYT?

Reply:
good idea! added in https://github.com/woocommerce/woocommerce-ios/pull/2802/commits/5dd038da898e1407f480e277ede5d555eababe9e